### PR TITLE
`AirbyteHook` add cancel job option

### LINKED ISSUE #24574: `AirbyteHook` add cancel job option
### Apache Airflow Provider(s)

airbyte

### Versions of Apache Airflow Providers

I want to cancel the job if it running more than specific time . Task is getting timeout however, airbyte job was not cancelled. it seems, on kill feature has not implemented 

Workaround:
  Create a custom operator and implement cancel hook and call it in on kill function. 

   def on_kill(self):
        if (self.job_id):
            self.log.error('on_kill: stopping airbyte Job %s',self.job_id)
            self.hook.cancel_job(self.job_id)

### Apache Airflow version

2.0.2

### Operating System

Linux

### Deployment

MWAA

### Deployment details

Airflow 2.0.2

### What happened

airbyte job was not cancelled upon timeout 

### What you think should happen instead

it should cancel the job

### How to reproduce

Make sure job runs more than timeout

sync_source_destination = AirbyteTriggerSyncOperator(
                    task_id=f'airbyte_{key}',
                    airbyte_conn_id='airbyte_con',
                    connection_id=key,
                    asynchronous=False,
                    execution_timeout=timedelta(minutes=2)
            )

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by sivankumar86**
closes: https://github.com/apache/airflow/issues/24574

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/airbyte/operators/airbyte.py
Comment (uranusjr) [inline airflow/providers/airbyte/operators/airbyte.py]: This should probably be a WARNING or even just INFO instead? Cancelling a job doesn’t seem like an error to me. We don’t have a consistent rule on how `on_kill` should log, but most operators I checked log an INFO here, or even not at all.

**Review by uranusjr — COMMENTED** [review]
Also please fix static check errors.

**Review by uranusjr — DISMISSED** [review]

**Comment by sivankumar86** [comment]
@uranusjr Thank you for reviewing the changes and feedback. Could you review the latest commit ?

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
No; we need a second reviewer’s approval here to merge (project policy), so all we can do for now is to wait.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
> No; we need a second reviewer’s approval here to merge (project policy), so all we can do for now is to wait.

Actually @uranusjr  - the policy is only for core changes - not providers. Providers are cool to merge with one committer. I often do so - anyhow there is always additional step of verification at release candidate time when I ask people to test their changes so we have this extra "gate" before release.

@sivankumar86 - you actually need to fix things. static checks and test are failing and it seems relevant to your changes. Install pre-commit and it will fix the static checks for you mostly.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Timeout on sdist building has been fixed in main already. Merging.

**Pull request merged**
